lib/web_ui/lib/src/engine/browser_location.dart

```diff
@@ -121,6 +115,63 @@ class HashLocationStrategy extends LocationStrategy {
   }
 }
 
+/// This is an implementation of [LocationStrategy] that uses the browser URL's
+/// [path](https://en.wikipedia.org/wiki/Uniform_Resource_Locator#Syntax)
+/// to represent its state.
+///
+/// In order to use this [LocationStrategy] for an app, it needs to be set in
+/// [engine.window.locationStrategy.locationStrategy]:
+class PathLocationStrategy extends LocationStrategy {
```

Comment:
This seems to have a lot of duplicated code from the hashlocationstrategy. We should probably refactor out a mixin